### PR TITLE
Update Go to 1.18.3, golangci-lint to 1.46.2

### PR DESCRIPTION
**Pull request by tklauser**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ TEST_TIMEOUT ?= 5s
 RELEASE_UID ?= $(shell id -u)
 RELEASE_GID ?= $(shell id -g)
 
-GOLANGCILINT_WANT_VERSION = 1.46.0
```
Comment (sayboras) [inline Makefile:13]: should we bump this as well ?

https://github.com/cilium/cilium/blob/6a8fe290930f73da0311813259f93ac244b7004a/.github/workflows/lint-go.yaml#L58

**Review by sayboras — APPROVED** [review]
LGTM :100:

File: Makefile (hunk 1 above)
Comment (tklauser) [reply Makefile:13]: Good point, will do as part of https://github.com/cilium/cilium/pull/20061

Comment (sayboras) [reply Makefile:13]: oops, did I just review PR with wrong repo in mind :rofl: :face_exhaling:

Comment (tklauser) [reply Makefile:13]: :D fixed it up in this repo as well and repushed

**Review by michi-covalent — APPROVED** [review]

**Comment by michi-covalent** [comment]
@tklauser what do you think about using `go-version-file`? it's supposed to find the latest patch version based on go.mod file.

https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

**Review by nebril — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**